docker/templates/non_proxy_headers.conf.template

```diff
@@ -1,10 +1,11 @@
 # use INLINE_RUNTIME_CHECK=false yarn build and nonce / hash to apply a CSP
 # Use add_header Content-Security-Policy "......." always;
-add_header Content-Security-Policy "default-src 'none'; connect-src 'self' *.sentry.io *.s3.amazonaws.com; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' fonts.googleapis.com 'unsafe-inline'; img-src 'self' data: blob: *.s3.amazonaws.com; font-src 'self' fonts.gstatic.com; object-src 'none';" always;
+add_header Content-Security-Policy "default-src 'none'; connect-src 'self' *.sentry.io *.s3.amazonaws.com; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' fonts.googleapis.com 'unsafe-inline'; img-src 'self' data: blob: *.s3.amazonaws.com https://*.openstreetmap.org; font-src 'self' fonts.gstatic.com; object-src 'none';" always;
 
 add_header Referrer-Policy "strict-origin-when-cross-origin";
 add_header Strict-Transport-Security "max-age=63072000; includeSubdomains;" always;
 add_header X-Frame-Options "SAMEORIGIN" always;
 add_header X-XSS-Protection "1; mode=block" always;
 add_header X-Content-Type-Options "nosniff" always;
-add_header Feature-Policy "geolocation 'none'; midi 'none'; notifications 'none'; push 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'none'; fullscreen 'none'; payment 'none'; usb 'none';";
+add_header Feature-Policy "geolocation 'self'; midi 'none'; notifications 'none'; push 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'none'; fullscreen 'none'; payment 'none'; usb 'none';";
+add_header Permissions-Policy "geolocation=(self)";
```

Comment:
`Permissions-Policy` is the modern version of `Feature-Policy`.

I think it's right to put this change here in `non_proxy_headers.conf.template`. My only hesitation is that these allowances are now also included for `/assets`, which I don't _think_ they need to be. (It worked if I just put the add_header in the `location /` block and not `location /assets`). I don't want to grant additional allowances if not necessary. I am not clear on the risk though.

https://github.com/greenriver/hmis-frontend/blob/06ba81e97cb1f1903d8f6dd6216d692e979b85ca/docker/templates/default.conf.template#L33-L34

Reply:
@yakloinsteak Can you review this nginx change for me? I confirmed it's working on QA.

In summary:
1. Allow fetching images from `https://*.tile.openstreetmap.org`, to support rendering maps
2. Allow requesting geolocation from the browser